### PR TITLE
feat(fulfillment): Migration backward compatibility

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Update migration for backward compatibility. This does not take into account data migration and table cleanup (@olivermrbl do we have a general tasks for the modules on that subject)?
- Rename fulfillment provider id to provider id
- add integration tests to ensure the migration backward compatibility
- add new module type for the options to be used in the medusa config for example

FIXES CORE-1724

### CODE REVIEW

**Comment by adrien2p** [comment]
WIP until I add some integration tests

**Review by olivermrbl — APPROVED** [review]

**Comment by adrien2p** [comment]
@olivermrbl you ll have to re review once I push my changes 👍

File: integration-tests/modules/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "license": "MIT",
   "private": true,
   "scripts": {
-    "test:integration": "node --expose-gc ./../../node_modules/.bin/jest --ci --silent=false -i --detectOpenHandles --logHeapUsage --forceExit",
+    "test:integration": "node --expose-gc ./../../node_modules/.bin/jest --ci --silent=false -i --detectOpenHandles --logHeapUsage --forceExit -- __tests__/fulfillment/index.spec.ts",
```
Comment (olivermrbl) [inline integration-tests/modules/package.json]: todo: remove file path

File: packages/fulfillment/src/types/index.ts [hunk 2]
```diff
@@ -7,3 +12,22 @@ export type InitializeModuleInjectableDependencies = {
 
 export const FulfillmentIdentifiersRegistrationName =
   "fulfillment_providers_identifier"
+
+export type FulfillmentModuleServiceInitializeOptions =
```
Comment (olivermrbl) [inline packages/fulfillment/src/types/index.ts]: suggestion:
```suggestion
export type FulfillmentModuleOptions =
```

**Review by olivermrbl — APPROVED** [review]

File: integration-tests/modules/package.json (hunk 1 above)
Comment (adrien2p) [reply integration-tests/modules/package.json]: You probably have an old version 👍

**Pull request merged**